src/Parser/DeathKnight/Unholy/Modules/Features/DarkTransformationAndWounds.js

```diff
@@ -0,0 +1,123 @@
+import React from 'react';
+
+import SPELLS from 'common/SPELLS';
+import SpellIcon from 'common/SpellIcon';
+import SpellLink from 'common/SpellLink';
+
+import Analyzer from 'Parser/Core/Analyzer';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import StatisticBox, { STATISTIC_ORDER } from 'Main/StatisticBox';
+import { encodeTargetString } from 'Parser/Core/Modules/EnemyInstances';
+import Wrapper from 'common/Wrapper';
+
+class DarkTransformationAndWounds extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+  };
+  // used to track how many stacks a target has
+  targets = {};
+
+  darkTransformationActive = false;
+  darkTransformationCasts = 0;
+  stopWatchActive = false;
+  stopWatch = 0;
+  totalTime = 0;
+  
+  
+  on_byPlayer_applybuff(event){
+	  const spellId = event.ability.guid;
+	  if(spellId === SPELLS.DARK_TRANSFORMATION.id){
+		  this.darkTransformationActive = true;
+		  this.darkTransformationCasts += 1;
+	  }
+  }
+  
+  on_byPlayer_removebuff(event){
+	  const spellId = event.ability.guid;
+	  if(spellId === SPELLS.DARK_TRANSFORMATION.id){
+		  this.darkTransformationActive = false;
+	  }
+  }
+  
+  on_byPlayer_applydebuffstack(event){
+    const spellId = event.ability.guid;
+    if(spellId === SPELLS.FESTERING_WOUND.id){
+		  this.targets[encodeTargetString(event.targetID, event.targetInstance)] = event.stack;
+	  }
+  }
+
+  on_byPlayer_removedebuffstack(event){
+    const spellId = event.ability.guid;
+    if(spellId === SPELLS.FESTERING_WOUND.id){
+		  this.targets[encodeTargetString(event.targetID, event.targetInstance)] = event.stack;
+    }
+  }
+
+  on_byPlayer_removedebuff(event){
+    const spellId = event.ability.guid;
+    if(spellId === SPELLS.FESTERING_WOUND.id){
+		  this.targets[encodeTargetString(event.targetID, event.targetInstance)] = 0;
+	  }
+  }
+
+  on_byPlayer_cast(event){
+    if(this.targets.hasOwnProperty(encodeTargetString(event.targetID, event.targetInstance)) && this.targets[encodeTargetString(event.targetID, event.targetInstance)] === 0) {
```

Comment:
This wont catch the case where your target has never had any stacks and thus aint in the array. If you instead invert the first check and use `||` instead of `&&` it will do what you want:
not in array -> first part of the condition evaluates to true and executes the body without evaluating the second part of the if condition.
in array -> first part of the condition evaluates to false and it continues to second part and executes the body if stacks === 0.
```
if(!this.targets.hasOwnProperty(encodeTargetString(event.targetID, event.targetInstance)) 
|| this.targets[encodeTargetString(event.targetID, event.targetInstance)] === 0)
```